src/catalog/src/local/memory.rs

```diff
@@ -87,6 +87,24 @@ impl CatalogManager for MemoryCatalogManager {
             .map(|v| if v.is_some() { 0 } else { 1 })
     }
 
+    async fn deregister_table(&self, request: DeregisterTableRequest) -> Result<usize> {
+        let catalogs = self.catalogs.write().unwrap();
+        let catalog = catalogs
+            .get(&request.catalog)
+            .context(CatalogNotFoundSnafu {
+                catalog_name: &request.catalog,
+            })?
+            .clone();
+        let schema = catalog
+            .schema(&request.schema)?
+            .with_context(|| SchemaNotFoundSnafu {
+                schema_info: format!("{}.{}", &request.catalog, &request.schema),
+            })?;
+        schema
+            .deregister_table(&request.table_name)
+            .map(|v| if v.is_some() { 0 } else { 1 })
```

Comment:
Looks like we should return `1` if `v.is_some()`.

BTW, should we change the return value to `bool`?

Reply:
I prefer bool

Reply:
You can also change return value of `register_table`  and `register_schema` to bool

Reply:
I think `Result<()>` is enough for register operations, they do not need to tell if the things already exist. And if we treat deregistering vacant things as an error, `deregister_table` can also be changed to return `Result<()>`

Reply:
I'm not sure whether some places use the return value to implement things like `CREATE IF NOT EXISTS`. If there aren't, `Result<()>` should be enough.

Reply:
`CREATE IF NOT EXISTS` is handled by table engine. On catalog manager we can return OK on both the new entry is newly inserted or the new entry that already exist. My thought is the registry manager only needs to ensure some entry is registered, or some entry doesn't exist. So wrap situations that already exist in registering or don't exist in deregistering look fine.

Reply:
In case this information is needed in the future, I'll keep a bool in the return type.